src/stub/core_impl/tracepoints.rs

```diff
@@ -0,0 +1,304 @@
+                                .ok_or(Error::TargetMismatch)?;
+                        ops.tracepoint_create(new_tracepoint).handle_error()?
+                    }
+                    QTDP::Define(dtdp) => {
+                        let define_tracepoint =
+                            DefineTracepoint::<<T::Arch as Arch>::Usize>::from_tdp(dtdp)
+                                .ok_or(Error::TargetMismatch)?;
+                        ops.tracepoint_define(define_tracepoint).handle_error()?
+                    }
+                };
+                // TODO: support qRelocInsn?
+                return Ok(HandlerStatus::NeedsOk);
+            }
+            Tracepoints::QTBuffer(buf) => {
+                match buf {
+                    QTBuffer::Request {
+                        offset,
+                        length,
+                        data,
+                    } => {
+                        let read = ops
+                            .trace_buffer_request(offset, length, data)
+                            .handle_error()?;
+                        if let Some(read) = read {
+                            let read = read.min(data.len());
+                            res.write_hex_buf(&data[..read])?;
+                        } else {
+                            res.write_str("l")?;
+                        }
+                    }
+                    QTBuffer::Configure { buffer } => {
+                        ops.trace_buffer_configure(buffer).handle_error()?;
+                    }
+                }
+                // Documentation doesn't mention this, but it needs OK
+                return Ok(HandlerStatus::NeedsOk);
+            }
+            Tracepoints::QTStart(_) => {
+                ops.trace_experiment_start().handle_error()?;
+                // Documentation doesn't mention this, but it needs OK
+                // TODO: qRelocInsn reply support?
+                return Ok(HandlerStatus::NeedsOk);
+            }
+            Tracepoints::QTStop(_) => {
+                ops.trace_experiment_stop().handle_error()?;
+                // Documentation doesn't mention this, but it needs OK
+                return Ok(HandlerStatus::NeedsOk);
+            }
+            Tracepoints::QTFrame(req) => {
+                let parsed_qtframe: Option<FrameRequest<<T::Arch as Arch>::Usize>> = req.0.into();
+                let parsed_req = parsed_qtframe.ok_or(Error::TargetMismatch)?;
+                let mut err: Result<_, Error<T::Error, C::Error>> = Ok(());
+                let mut any_results = false;
+                ops.select_frame(parsed_req, &mut |desc| {
+                    // TODO: bubble up the C::Error from this closure? list_thread_active does this
+                    // instead
+                    let e = (|| -> Result<_, _> {
+                        match desc {
+                            FrameDescription::FrameNumber(n) => {
+                                res.write_str("F ")?;
```

Comment:
I definitely missed that section of the docs! This codepath *does* actually work, however: in the `armv4t` example that's included in this MR I can do
```
(gdb) tfind 3
Found trace frame 3, tracepoint 1
#0  main () at test.c:10
10	in test.c
(gdb) tfind 2
Found trace frame 2, tracepoint 1
10	in test.c
```

Reply:
Oh GDB... thou art truly exceptional software... /s

If we peek inside GDB and look at how it parses the packet response, we find this code

```c
      case 'F':
	p = ++reply;
	target_frameno = (int) strtol (p, &reply, 16);
```

(via https://github.com/bminor/binutils-gdb/blob/master/gdb/remote.c#L14258-L14263)

And lo, reading [the docs](https://en.cppreference.com/w/c/string/byte/strtol) of `strtol` reveals this lovely feature:

> Discards any whitespace characters (as identified by calling [isspace](https://en.cppreference.com/w/c/string/byte/isspace)) until the first non-whitespace character is found [...]

So that solves that mystery...

In any case, even though this "works", lets make sure `gdbstub` remains spec-compliant, in case other GDB RSP clients aren't quite so forgiving of bonus whitespace here.

* * *

Apologies for assuming that you hadn't tested this code, hopefully you understand why I might've gotten that assumption from reading just this code + the corresponding spec 😅